### PR TITLE
ROX-13811: Enable Performance Insights for managed DBs

### PR DESCRIPTION
**Pull request by vladbologa**
## Description

We'd like to enable Performance Insights (an AWS RDS feature) for the first users of Field Trial, to get a better idea of DB usage.

With this PR, the feature can be enabled using an environment variable for fleetshard-sync. Perhaps in the future we could allow it to be enabled on a per-customer basis (e.g. in the admin API)

## Checklist (Definition of Done)

~~- [ ] Unit and integration tests added~~
~~- [ ] Added test description under `Test manual`~~
~~- [ ] Evaluated and added CHANGELOG.md entry if required~~
~~- [ ] Documentation added if necessary (i.e. changes to dev setup, test execution, ...)~~
- [ ] CI and all relevant tests are passing

~~- [ ] Add the ticket number to the PR title if available, i.e. `ROX-12345: ...`~~
~~- [ ] Discussed security and business related topics privately. Will move any security and business related topics that arise to private communication channel.~~

## Test manual

Tested in a local cluster, with the RDS database provisioned in a dev account.
```
# run the operator locally (from stackrox/operator)
make install run

# all other commands are run from the acs-fleet-manager root dir
# bootstrap the test env
INSTALL_OPERATOR=NO INSTALL_OLM=NO ./dev/env/scripts/bootstrap.sh

# bring up fleet-manager and fleet-shard
INSTALL_OPERATOR=NO MANAGED_DB_ENABLED=TRUE MANAGED_DB_SUBNET_GROUP=acs-dev-dp-01-subnet-group-1 MANAGED_DB_SECURITY_GROUP=sg-066eb99769c899e79 MANAGED_DB_PERFORMANCE_INSIGHTS=TRUE ./dev/env/scripts/up.sh

# create a test central
./scripts/create-central.sh

# verify that the Central is up and running
# verify in the AWS console that the DB is being created, Perf Insights are enabled, and is finally deleted

#delete test central
curl -X DELETE -H "Authorization: Bearer $(ocm token)" -H "Content-Type: application/json" \
  http://127.0.0.1:8000/api/rhacs/v1/centrals/cdvq4mmsbpn001pjdfkg\?async\=true
```

### CODE REVIEW

**Review by ebensh — APPROVED** [review]

File: fleetshard/pkg/central/cloudprovider/awsclient/rds.go [hunk 1]
```diff
@@ -23,23 +23,29 @@ const (
...
-	dbName           = "postgres"
 	awsRetrySeconds  = 30
+
+	// DB cluster / instance configuration parameters
+	dbEngine                = "aurora-postgresql"
+	dbEngineVersion         = "13.7"
+	dbInstanceClass         = "db.serverless"
+	dbPostgresPort          = 5432
+	dbName                  = "postgres"
+	dbMinCapacity           = 0.5
```
Comment (ebensh) [inline fleetshard/pkg/central/cloudprovider/awsclient/rds.go:38]: Maybe specify units here - vCPU? Days?

File: dev/env/manifests/fleetshard-sync/02-fleetshard-sync-deployment.yaml [hunk 2]
```diff
@@ -49,6 +49,8 @@ spec:
               value: "$MANAGED_DB_SECURITY_GROUP"
             - name: MANAGED_DB_SUBNET_GROUP
               value: "$MANAGED_DB_SUBNET_GROUP"
+            - name: MANAGED_DB_PERFORMANCE_INSIGHTS
+              value: "$MANAGED_DB_PERFORMANCE_INSIGHTS"
```
Comment (kovayur) [inline dev/env/manifests/fleetshard-sync/02-fleetshard-sync-deployment.yaml:53]: Aren't we going to include this parameter to the helm chart as well?

File: fleetshard/pkg/central/cloudprovider/awsclient/rds.go (hunk 1 above)
Comment (vladbologa) [reply fleetshard/pkg/central/cloudprovider/awsclient/rds.go:38]: It's ACUs - Aurora Capacity Unit
1 ACU = 1 vCPU + 2GB RAM

Should it be like `dbMinCapacityACU`? and a comment describing the acronym?

File: dev/env/manifests/fleetshard-sync/02-fleetshard-sync-deployment.yaml (hunk 2 above)
Comment (vladbologa) [reply dev/env/manifests/fleetshard-sync/02-fleetshard-sync-deployment.yaml:53]: yes, doing that now

**Review by kovayur — CHANGES_REQUESTED** [review]
Pls add `MANAGED_DB_PERFORMANCE_INSIGHTS` parameter to the helm chart and enable it

File: dp-terraform/helm/rhacs-terraform/templates/fleetshard-sync.yaml [hunk 3]
```diff
@@ -62,6 +62,8 @@ spec:
           value: {{ required "fleetshardSync.managedDB.subnetGroup is required when fleetshardSync.managedDB.enabled = true" .Values.fleetshardSync.managedDB.subnetGroup }}
         - name: MANAGED_DB_SECURITY_GROUP
           value: {{ required "fleetshardSync.managedDB.securityGroup is required when fleetshardSync.managedDB.enabled = true" .Values.fleetshardSync.managedDB.securityGroup }}
+        - name: MANAGED_DB_PERFORMANCE_INSIGHTS
+          value: { { .Values.fleetshardSync.managedDB.performanceInsights | quote } }
```
Comment (kovayur) [inline dp-terraform/helm/rhacs-terraform/templates/fleetshard-sync.yaml:66]:
```suggestion
          value: {{ .Values.fleetshardSync.managedDB.performanceInsights | quote }}
```

**Review by kovayur — CHANGES_REQUESTED** [review]
All good except the small typo

**Pull request merged**